src/components/tables/Details/StatusIndicatorAndLabel.tsx

```diff
@@ -0,0 +1,56 @@
+import { TableCell, Typography } from '@mui/material';
+import { Shard } from 'data-plane-gateway/types/shard_client';
+import { useRouteStore } from 'hooks/useRouteStore';
+import { FormattedMessage } from 'react-intl';
+import { shardDetailSelectors } from 'stores/ShardDetail';
+
+interface Props {
+    shard: Shard;
+}
+
+function StatusIndicatorAndLabel({ shard }: Props) {
+    const { id } = shard.spec;
+
+    const shardDetailStore = useRouteStore();
```

Comment:
This needs to be renamed as a hook like the files above.

Reply:
PENDING.